crates/distribution-types/src/file.rs

```diff
@@ -53,12 +53,10 @@ impl File {
             size: file.size,
             upload_time_utc_ms: file.upload_time.map(|dt| dt.timestamp_millis()),
             url: if file.url.contains("://") {
-                let url = safe_copy_url_auth_to_str(base, &file.url)
-                    .map_err(|err| FileConversionError::Url(file.url.clone(), err))?
-                    .map(|url| url.to_string())
-                    .unwrap_or(file.url);
-
-                FileLocation::AbsoluteUrl(url)
+                let url = Url::parse(&file.url)
+                    .map_err(|err| FileConversionError::Url(file.url.clone(), err))?;
+                let url = AuthenticationStore::with_url_encoded_auth(url);
```

Comment:
So the assumption here is that we've already hit the base URL on a previous invocation of the middleware, and so its credentials are already stored in the `AuthenticationStore`, and will be propagated here if needed?

Reply:
Yes, that was my assumption.  That if `base` has auth, it received it from a copy from a previous request (in `FlatIndexClient.read_from_url` or `RegistryClient.simple_single_index`).